### PR TITLE
Additional update the ProfilePage

### PR DESCRIPTION
**Pull request by MaksimkaSota**
Feature:
1. Added updatePhoto logic for ProfilePage;
2. Added updateData logic for ProfilePage.

Refactor:
1. Edited mixins;
2. Added common components for forms.

Server errors will be handled at a later stage of development!

### CODE REVIEW

File: src/Components/Main/Pages/ProfilePage/ContactForm/ContactForm.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  return (
+    <div className={classes.contactBlock}>
+      <div className={classes.contact}>
+        <b className={classes.title}>{title}:</b>
+        <FormField
+          classNameField={cn(classes.field, {
+            [classes.fieldError]: status && status.contacts && status.contacts[title]
+          })}
+          name={name}
+          type={'text'}
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/ProfilePage/ContactForm/ContactForm.js]:
```suggestion
          type="text"
```

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <b className={classes.title}>Full name:</b>
+        <FormField classNameField={cn(classes.field, {[classes.fieldError]: status && status['fullName']})}
+                   name="fullName"
+                   type="text"
+                   placeholder="Full name"
+                   onChange={handleChange}
+                   errors={errors} />
+      </div>
+      {
+        (status && status['fullName']) &&
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]:
```suggestion
        status?.fullName &&
```

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          <b className={classes.title}>My professional skills:</b>
+          <FormField classNameField={cn(classes.textarea, {[classes.fieldError]: status && status['Job']})}
+                     name="lookingForAJobDescription"
+                     component="textarea"
+                     placeholder="My professional skills"
+                     onChange={handleChange} />
+        </div>
+      }
+      {
+        (status && status['Job']) &&
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]:
```suggestion
        status?.Job &&
```

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      <div className={classes.descriptionBlock}>
+        <b className={classes.title}>About me:</b>
+        <FormField classNameField={cn(classes.textarea, {[classes.fieldError]: status && status['aboutMe']})}
+                   name="aboutMe"
+                   component="textarea"
+                   placeholder="About me"
+                   onChange={handleChange} />
+      </div>
+      {
+        (status && status['aboutMe']) &&
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]:
```suggestion
        status?.aboutMe &&
```

File: src/Components/Main/Pages/ProfilePage/ProfileInfo/ProfileInfoContainer.js [hunk 5]
```diff
@@ -2,26 +2,47 @@ import { ProfileInfo } from './ProfileInfo';
...
     (status) => dispatch(updateStatus(status)),
     [dispatch]
   );
+  const updatePhotoCallback = useCallback(
+    (photo) => dispatch(updatePhoto(photo)),
+    [dispatch]
+  );
+  const updateDataCallback = useCallback(
+    (profileData, setStatus, setSubmitting, setEditModeData) => {
+      return dispatch(updateData(profileData, setStatus, setSubmitting, setEditModeData));
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/ProfilePage/ProfileInfo/ProfileInfoContainer.js]: Do you really need "return" here? =)

**Review by ZhekaVasil — APPROVED** [review]

File: src/Components/Main/Pages/ProfilePage/ContactForm/ContactForm.js (hunk 1 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/ProfilePage/ContactForm/ContactForm.js]: Done!

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js (hunk 2 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]: Done!

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js (hunk 3 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]: Done!

File: src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js (hunk 4 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/ProfilePage/ProfileDataForm/ProfileDataForm.js]: Done!

File: src/Components/Main/Pages/ProfilePage/ProfileInfo/ProfileInfoContainer.js (hunk 5 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/ProfilePage/ProfileInfo/ProfileInfoContainer.js]: No, this is not necessary.
I have removed the "return" !

**Pull request merged**